opamp/observiq/observiq_client.go

```diff
@@ -118,24 +127,30 @@ func NewClient(args *NewClientArgs) (opamp.Client, error) {
 
 func (c *Client) addManagedConfigs(args *NewClientArgs) error {
 	// Add configs to config manager
-	managerManagedConfig, err := opamp.NewManagedConfig(args.ManagerConfigPath, managerReload(c, args.ManagerConfigPath))
+	managerManagedConfig, err := opamp.NewManagedConfig(args.ManagerConfigPath, managerReload(c, args.ManagerConfigPath), true)
 	if err != nil {
 		return fmt.Errorf("failed to create manager managed config: %w", err)
 	}
 	c.configManager.AddConfig(ManagerConfigName, managerManagedConfig)
 
-	collectorManagedConfig, err := opamp.NewManagedConfig(args.CollectorConfigPath, collectorReload(c, args.CollectorConfigPath))
+	collectorManagedConfig, err := opamp.NewManagedConfig(args.CollectorConfigPath, collectorReload(c, args.CollectorConfigPath), true)
 	if err != nil {
 		return fmt.Errorf("failed to create collector managed config: %w", err)
 	}
 	c.configManager.AddConfig(CollectorConfigName, collectorManagedConfig)
 
-	loggerManagedConfig, err := opamp.NewManagedConfig(args.LoggerConfigPath, loggerReload(c, args.LoggerConfigPath))
+	loggerManagedConfig, err := opamp.NewManagedConfig(args.LoggerConfigPath, loggerReload(c, args.LoggerConfigPath), true)
 	if err != nil {
 		return fmt.Errorf("failed to create logger managed config: %w", err)
 	}
 	c.configManager.AddConfig(LoggingConfigName, loggerManagedConfig)
 
+	reportManagedConfig, err := opamp.NewManagedConfig("report.yaml", reportReload(c), false)
+	if err != nil {
+		return fmt.Errorf("failed to create report managed config: %w", err)
+	}
+	c.configManager.AddConfig(ReportConfigName, reportManagedConfig)
```

Comment:
Potential refactor we could do here - make an interface that `ManagedConfig` satisfies and have a `MemoryManagedConfig` that also implements it. I think it would make this more clear that this config is different and we could avoid having to do things like put a bogus file name here.

Reply:
not sure how cleanly this can actually be done but 👍 from me

Reply:
Yeah, I agree. This is definitely a punching holes in stuff to get things to work approach. I'll write a ticket to revisit this.

Reply:
This may go away as there is talk of finding a way to send custom messages via opamp which would remove the need for this.

Reply:
I just did a quick look and I think we can do as @BinaryFissionGames suggested and make an interface for ManagedConfig. I'm going to hold off for now as we are really currently abusing the system to send a custom command. It would be ideal for OPAMP to have a way to do this.